src/index.js

```diff
@@ -0,0 +1,22 @@
+const fs = require("fs");
+const { colors } = require("@primer/primitives");
+
+const getTheme = require("./theme");
+
+/* The array of light to dark colors are reversed to auto-generate dark theme */
+const darkColors = {};
+Object.entries(colors).forEach(([name, val]) => {
+  if (name === "black") {
+    darkColors.white = val;
+  } else if (name === "white") {
+    darkColors.black = val;
+  } else {
+    darkColors[name] = [...val].reverse();
```

Comment:
This is the key part. The colors array go from light to dark shades. For the dark theme, we simply reverse the ordering. This works surprisingly well!